MYSQL.txt

```diff
@@ -0,0 +1,2075 @@
+    -> FROM Salespeople INNER JOIN Orders 
+    -> ON Salespeople.SNUM=Orders.SNUM
+    -> GROUP BY Orders.SNUM,Orders.ODATE;
+
++---------+------------+-----------+
+| SNAME   | ODATE      | MAX_ORDER |
++---------+------------+-----------+
+| Rifkin  | 1990-10-03 |   1098.16 |
+| Motika  | 1990-10-03 |    1900.1 |
+| Peel    | 1990-10-03 |    767.19 |
+| Serres  | 1990-10-03 |   5160.45 |
+| Serres  | 1990-10-04 |     75.75 |
+| Peel    | 1990-10-05 |      4723 |
+| AxelRod | 1990-10-04 |   1713.23 |
+| Serres  | 1990-10-06 |   1309.95 |
+| Peel    | 1990-10-06 |   9891.88 |
++---------+------------+-----------+ 
+
+
+4) Arrange the Order table by descending customer number. 
+
+mysql> SELECT ONUM,AMT,ODATE,CNUM,SNUM FROM Orders 
+    -> ORDER BY CNUM DESC;
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-10-03 | 2008 | 1007 |
+| 3006 | 1098.16 | 1990-10-03 | 2008 | 1007 |
+| 3002 |  1900.1 | 1990-10-03 | 2007 | 1004 |
+| 3008 |    4723 | 1990-10-05 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-10-06 | 2006 | 1001 |
+| 3007 |   75.75 | 1990-10-04 | 2004 | 1002 |
+| 3010 | 1309.95 | 1990-10-06 | 2004 | 1002 |
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-10-04 | 2002 | 1003 |
+| 3003 |  767.19 | 1990-10-03 | 2001 | 1001 |
++------+---------+------------+------+------+
+ 
+
+5) Find which salespeople currently have orders in the order table. 
+
+mysql> SELECT SNUM,SNAME FROM Salespeople
+    -> WHERE
+    -> SNUM IN(SELECT DISTINCT(SNUM) FROM Orders);
++------+---------+
+| SNUM | SNAME   |
++------+---------+
+| 1001 | Peel    |
+| 1002 | Serres  |
+| 1003 | AxelRod |
+| 1004 | Motika  |
+| 1007 | Rifkin  |
++------+---------+
+  
+
+6) List names of all customers matched with the salespeople serving them. 
+
+mysql> SELECT Customers.CNUM,Customers.CNAME,Salespeople.SNUM,Salespeople.SNAME
+    -> FROM Salespeople JOIN Customers
+    -> WHERE
+    -> Salespeople.snum=Customers.SNUM;
+
++------+----------+------+---------+
+| CNUM | CNAME    | SNUM | SNAME   |
++------+----------+------+---------+
+| 2001 | Hoffman  | 1001 | Peel    |
+| 2002 | Giovanni | 1003 | AxelRod |
+| 2003 | Liu      | 1002 | Serres  |
+| 2004 | Grass    | 1002 | Serres  |
+| 2006 | Clemens  | 1001 | Peel    |
+| 2008 | Cisneros | 1007 | Rifkin  |
+| 2007 | pereira  | 1004 | Motika  |
++------+----------+------+---------+  
+
+
+ 7) Find the names and numbers of all salespeople who have more than one customer. 
+   
+ mysql> SELECT SNUM , SNAME  FROM Salespeople 
+     -> WHERE 
+     -> SNUM IN(SELECT SNUM  FROM Customers 
+     -> GROUP BY SNUM 
+     -> HAVING 
+     -> COUNT(SNUM) > 1) ;
+    
++------+--------+
+| SNUM | SNAME  |
++------+--------+
+| 1001 | Peel   |
+| 1002 | Serres |
+
+
+8) Count the orders of each of the salespeople and output the results in descending order. 
```

Comment:
Try to think if there are salespeople who are not associated with any orders then their order count should also be displayed which is 0.